src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/TestFileMarkupParser.cs

```diff
@@ -204,6 +204,10 @@ public static void GetPositionAndSpans(string input, out int? cursorPositionOpt,
         public static void GetPositionAndSpans(string input, out string output, out int cursorPosition, out IDictionary<string, IList<TextSpan>> spans)
         {
             GetPositionAndSpans(input, out output, out int? cursorPositionOpt, out spans);
+            if (cursorPositionOpt is null)
+            {
+                throw new InvalidOperationException();
```

Comment:
Can we provide a better error message here to the user?

Reply:
Done!